search/localsearch_nqueens.cpp

```diff
@@ -0,0 +1,170 @@
+			for (size_t i = 0; i < solution_vector.size(); i++) {
+				for (size_t j = i + 1; j < solution_vector.size(); j++) {
+					if ((solution_vector[i] - (i + 1))
+							== (solution_vector[j] - (j + 1))) {
+						h++;
+						break;
+					}
+				}
+			}
+			//sum inverse diagonal
+			for (size_t i = 0; i < solution_vector.size(); i++) {
+				for (size_t j = i + 1; j < solution_vector.size(); j++) {
+					if ((solution_vector[i] + (i + 1))
+							== (solution_vector[j] + (j + 1))) {
+						h++;
+						break;
+					}
+				}
+			}
+			return h;
+		}
+
+		/**
+		 * function: neighbour
+		 * Generate a neighbor based on the current solution of
+		 * the problem. Read about neighborhoods in local search.
+		 * @param [in] solution_vector is a vector with the
+		 * column number where the queens are located on the board.
+		 * @returns a new vector similar to the one passed by parameter.
+		 */
+		std::vector<int> neighbour(std::vector<int> solution_vector) {
+			int64_t i = rand() % solution_vector.size() + 1;
+			int64_t mov = (rand() % solution_vector.size() + 1);
+			solution_vector[i - 1] = mov;
+			return solution_vector;
+		}
+
+		/**
+		 * function: print_solution
+		 * Prints the solution obtained on the chessboard in console.
+		 * @param [in] solution_vector is a vector with the
+		 * column number where the queens are located on the board.
+		 */
+		void print_solution(const std::vector<int> &solution_vector) {
+			for (int i = 0; i < solution_vector.size(); i++) {
+				for (int j = 0; j < solution_vector.size(); j++) {
+					if (j == solution_vector[i] - 1) {
+						std::cout << " Q ";
+					} else {
+						std::cout << " _ ";
+					}
+				}
+				std::cout << "\n";
+			}
+
+			std::cout << "\nSolucion: ";
+			for (int x : solution_vector) {
+				std::cout << x << " ";
+			}
+			std::cout << "\nCost:" << cost(solution_vector) << std::endl;
+		}
+
+		/**
+		 * function: searchrepeat
+		 * Search for repeated neighbors (vector) to
+		 * ensure the generation of new neighbors.
+		 * @param [in] prob_neighbor is candidate to be a neighbor of the current solution.
+		 * @param [in] neighborhood is vector of generated neighbors contained in memory.
+		 * @return true if value exists, if not false.
+		 */
+		inline bool searchrepeat(const std::vector<int> &prob_neighbor, const std::vector<std::vector<int>> &neighborhood) {
+			return find(neighborhood.begin(), neighborhood.end(), prob_neighbor) != neighborhood.end();
+		}
+	} //namespace localsearch
+} //namespace search
+
+/**
+ * Main function
+ * Heuristic Local Search - Greedily take the first neighbor that reduces the cost
+ * - s = generate initial solution
+ * - while s not optimal local do{
+ * -	s* belongs neighbor(s) with cost(s*) < cost(s)
+ * -	keep the neighboring solution if it has lower cost than the current one.
+ * -	s = s*
+ * -}
+ *
+ */
+int main() {
+	constexpr int nqueens = 8;  //The number of queens on the table is the same as the dimension of the chessboard.

```

Comment:
Preferably, add a space after the `//`.
Same in other parts of the code (below and above).